quantum/matrix.c

```diff
@@ -122,7 +132,7 @@ __attribute__((weak)) void matrix_read_cols_on_row(matrix_row_t current_matrix[]
     for (uint8_t col_index = 0; col_index < MATRIX_COLS; col_index++, row_shifter <<= 1) {
         pin_t pin = direct_pins[current_row][col_index];
         if (pin != NO_PIN) {
-            current_row_value |= readPin(pin) ? 0 : row_shifter;
+            current_row_value |= readPin(pin) ? PRESSED_KEY_PIN_STATE : row_shifter;
```

Comment:
In order to be somewhat correct, this should look like
```suggestion
            current_row_value |= (readPin(pin) == PRESSED_KEY_PIN_STATE) ? 0 : row_shifter;
```
However, I think that a better way to fix this is to update the direct pins code to use `readMatrixPin()`, and then change the implementation of `readMatrixPin()` to invert the pin state there (this will also fix the `NO_PIN` handling).